tidy3d/components/geometry.py

```diff
@@ -65,11 +65,16 @@ def point_inside(x: float, y: float, z: float):
             loc = Point(x, y)
             return any(shape.contains(loc) for shape in shapes_intersect)
 
-        x_arr, y_arr, z_arr = np.array(x), np.array(y), np.array(z)
-        inside = np.zeros((x_arr.size,), dtype=bool)
-        for ipt, (xp, yp, zp) in enumerate(zip(x_arr.ravel(), y_arr.ravel(), z_arr.ravel())):
-            inside[ipt] = point_inside(xp, yp, zp)
-        return inside.reshape(x_arr.shape)
+        arrays = tuple(map(np.array, (x, y, z)))
+        shapes = set(arr.shape for arr in arrays)
+        if len(shapes) > 1:
+            raise ValueError("all coordinate inputs (x,y,z) must be same shape.")
```

Comment:
This check has to be added to all `inside` methods that are overwritten then, right?

Could be a deocrator... Is there a way to use a classmethod or a staticmethod to decorate methods of the same class?

Also: note first capital letter in error message :P

Reply:
I think it needs to be either a decorator defined somewhere else (outside of the class), or just a regular method `check_equal_dims` that is called inside the other methods. What do you think?

Reply:
yea it might be, I think what you propose sounds good. maybe a separate function kind of like `ensure_freq_in_range` in medium.py